pdns/dnsdist.cc

```diff
@@ -1942,13 +1942,12 @@ static void healthChecksThread()
   setThreadName("dnsdist/healthC");
 
   constexpr int interval = 1;
+  auto states = g_dstates.getLocal(); // this points to the actual shared_ptrs!
```

Comment:
I am wondering how this works when the set of downstreams changes.

Reply:
We get a `LocalStateHolder` which holds a reference-counted snapshot of the last data, and checks whether there is an updated version every time we access the data, so it should be fine?

Reply:
Right. Somehow I thought the local state was only checked on `getLocal()`.